### DISCUSSION TITLE
how to extends FileNavigatorTree in theia

### DISCUSSION THREAD

**Discussion by bettyTheiaQues** [eclipse-theia/theia#9395] [Q&A]
I want to extend navigator to open a folder and display its contents，
i extend FileNavigatorWidget and FileNavigatorTree and use FileNavigatorModel.
When I click open folder, there is no folder content in navigator，i think i lack the implementation of resolvechildren, but I don't know how to achieve it. Is my idea right? Do you have any suggestions? Thank you very much.

**Comment by kittaakos** [comment]
> FileNavigatorModel.

Just a blind guess without seeing your code. Check if you have called `super.init()` from every place you have added a custom `init` method.

```ts
class YourFileNavigatorModel extends FileNavigatorModel {
  @postConstruct()
  init(): void {
   super.init();
   // your stuff
}
```

**Reply by bettyTheiaQues** [reply]
thanks,i have resolved the problem,now i have another question:
When I click the navigator file, how can I get the full path of the file or folder in the background？
Do you have any suggestions？

**Reply by kittaakos** [reply]
I have no idea what you're doing, but if your tree widget is initialized with

https://github.com/eclipse-theia/theia/blob/a0824730855cddf42f4ee6d59be9579184e850fb/packages/core/src/browser/tree/tree-widget.tsx#L101

then you can listen on

https://github.com/eclipse-theia/theia/blob/a0824730855cddf42f4ee6d59be9579184e850fb/packages/core/src/common/selection-service.ts#L32

check if your object has a URI and use the `FileService` to map the URI to a proper filesystem path

https://github.com/eclipse-theia/theia/blob/a0824730855cddf42f4ee6d59be9579184e850fb/packages/filesystem/src/browser/file-service.ts#L633

I hope this helps.

**Reply by bettyTheiaQues** [reply]
i will try it,thanks a lot
